### PR TITLE
don't crash when the layer indices don't line up right

### PR DESCRIPTION
**Pull request by grapesmoker**
If the interpolation offsets are missing, regsite crashes. This is bad.

### CODE REVIEW

File: regulations/generator/layers/layers_applier.py [hunk 1]
```diff
@@ -128,8 +128,12 @@ def get_layer_pairs(self, text_index, original_text):
...
             offset_locations = LocationReplace.find_all_offsets(o,
                                                                 original_text)
-            locations = [offset_locations.index(offset)]
-            layer_elements.append((o, r, locations))
+            try:
+                locations = [offset_locations.index(offset)]
+                layer_elements.append((o, r, locations))
+            except Exception as ex:
+                print ex
+                print offset_locations, offset
```
Comment (willbarton) [inline regulations/generator/layers/layers_applier.py]: Should we use Django logging here rather than just a print statement?

Comment (grapesmoker) [inline regulations/generator/layers/layers_applier.py]: Yeah, updated to reflect that.

Comment (willbarton) [inline regulations/generator/layers/layers_applier.py]: :+1:

**Comment by cmc333333** [comment]
Just a heads up that this also means your data is corrupted

**Comment by grapesmoker** [comment]
I wouldn't say "corrupted;" I'd say it doesn't conform to the assumptions that regsite is relying on. Which is bad and we need to know about it, but it's also not something that should bring down regsite entirely, which is what it's doing now. I'd much prefer to be notified about this in the console and be able to debug it at my leisure than have the entire thing go down because somewhere there's a reference that doesn't quite line up correctly.

**Comment by cmc333333** [comment]
I agree absolutely about not bringing down site. However, the fact that this error is arising is bad news for your data's consistency. Effectively, the layer data is saying "Replace the nth instance of this substring", but when it's trying to apply that logic, it can't find the nth substring. This means either the layer data doesn't match the tree or the logic in -site is removing one of these substrings before it should.

**Comment by grapesmoker** [comment]
Right, I understand what it's trying to do. And it is bad news, just not bad enough to stop everything. It's definitely not something we should be allowing to continue to happen, but the worst thing that occurs in this case is that some markup doesn't get inlined. That's not great, but it shouldn't be fatal.

**Comment by cmc333333** [comment]
:+1:

**Comment by willbarton** [comment]
batman-thumbs-up

**Pull request merged**